Update with new options

5axes opened the pull request:
As I have made the job for my own test .

But of course tyour are free to cancel this request

5axes committed:
Add New Options 

arcwelder_min_arc_segment
arcwelder_mm_per_arc_segment

arcwelder_allow_3d_arcs

5axes committed:
Add help file

help.txt

5axes committed:
Update README.md

5axes committed:
Update README.md

5axes committed:
Add comments

fieldOfView commented on plugin.json:
```diff
@@ -1,7 +1,7 @@
     "name": "Arc Welder plugin",
     "author": "fieldOfView",
-    "version": "3.5.0-DEV",
+    "version": "3.5.1-DEV",
```
Please revert this change. The version will remain 3.5.0-DEV until an official 3.5.0 version is published to the Marketplace.

fieldOfView commented on bin/win64/help.txt:
```diff
@@ -0,0 +1,75 @@
+
```
Please remove this file.

fieldOfView commented on README.md:
```diff
@@ -4,4 +4,6 @@ This plugin wraps [ArcWelderLib](https://github.com/FormerLurker/ArcWelderLib) b
 
-For further informatio about ArcWelder antistutter, see https://plugins.octoprint.org/plugins/arc_welder
\ No newline at end of file
+For further informations about ArcWelder antistutter, see https://plugins.octoprint.org/plugins/arc_welder
```
Thanks for ficing this typo, however `information` has on plural form in english. `For further information about ...`

fieldOfView commented on ArcWelderPlugin.py:
```diff
@@ -112,6 +113,14 @@ def _filterGcode(self, output_device):
         tolerance = global_container_stack.getProperty("arcwelder_tolerance", "value") / 100
         resolution = global_container_stack.getProperty("arcwelder_resolution", "value")
+        min_arc = int(global_container_stack.getProperty("arcwelder_min_arc_segment", "value"))
+        mm_par_arc = global_container_stack.getProperty("arcwelder_mm_per_arc_segment", "value")
```
Please change to `mm_per_arc_segment` (per, not par)

fieldOfView commented on ArcWelderPlugin.py:
```diff
@@ -112,6 +113,14 @@ def _filterGcode(self, output_device):
         maximum_radius = global_container_stack.getProperty("arcwelder_maximum_radius", "value")
         tolerance = global_container_stack.getProperty("arcwelder_tolerance", "value") / 100
         resolution = global_container_stack.getProperty("arcwelder_resolution", "value")
+        min_arc = int(global_container_stack.getProperty("arcwelder_min_arc_segment", "value"))
```
Please change to `min_arc_segment`

fieldOfView commented on ArcWelderPlugin.py:
```diff
@@ -112,6 +113,14 @@ def _filterGcode(self, output_device):
+        cmd_3d_arcs = ""
+        allow_3d_arcs = global_container_stack.getProperty("arcwelder_allow_3d_arcs", "value")
+        if allow_3d_arcs:
+            cmd_3d_arcs = "-z"
```
This could be written as 
`cmd_3d_arcs = "-z" if global_container_stack.getProperty("arcwelder_allow_3d_arcs", "value") else ""`

fieldOfView commented on ArcWelderPlugin.py:
```diff
@@ -134,8 +143,9 @@ def _filterGcode(self, output_device):
 
-                subprocess.run([arcwelder_path, "-m=%f" % maximum_radius, "-t=%f" % tolerance, "-r=%f" % resolution, path])
-
+                # Logger.log("d", "Process.run %s , %s, %s, %s, %s, %s, %s, %s", arcwelder_path, cmd_3d_arcs, "-m=%f" % maximum_radius, "-a=%d" % min_arc, "-s=%f" % mm_par_arc, "-t=%f" % tolerance, "-r=%f" % resolution, path)
```
Please remove the commented-out line.

fieldOfView commented on README.md:
```diff
@@ -4,4 +4,6 @@ This plugin wraps [ArcWelderLib](https://github.com/FormerLurker/ArcWelderLib) b
\ No newline at end of file
+For further informations about ArcWelder antistutter, see https://plugins.octoprint.org/plugins/arc_welder
+
+For further informations about MM Per Arc Segment and Min Arc Segments options, see https://github.com/FormerLurker/ArcWelderPlugin/issues/80
```
I'm not sure if we should link to specific issues for specific features. Ideally, the description of the setting would be good enough, and eventually I expect any documentation in that issue to trickle down to the official documentation of the arcwelder plugin.

fieldOfView commented on ArcWelderPlugin.py:
```diff
@@ -1,5 +1,6 @@
 # Copyright (c) 2020 Aldo Hoeben / fieldOfView
 # The ArcWelderPlugin for Cura is released under the terms of the AGPLv3 or higher.
+# 25/11/2020 add options arcwelder_min_arc_segment / arcwelder_mm_per_arc_segment / arcwelder_allow_3d_arcs
```
I don't keep a changelog inside my plugins.

fieldOfView reviewed (changes requested):
Thanks for the PR. I have some suggested changes.

5axes commented:
Ok finally too many differences between my code and what you would like to have. So I close the pull request.

5axes committed:
Changer parameter name

mm_per_arc
min_arc_segment

fieldOfView commented:
Hey @5axes, this is how pull requests work; I could also make these changes after merging in your version, but then you lay the burden with me. I don't mean to micromanage you, and appreciate your contribution, but the end-goal is to have a consistent codebase, with a consistent code style.

5axes committed:
Update option and Command line

Change value limits in the json file
Change the commande ligne if min_arc_segment is set

5axes committed:
change parameter name

mm_per_arc_segment

5axes committed:
Enable option on arcwelder_mm_per_arc_segment

arcwelder_enable and arcwelder_min_arc_segment>0

5axes committed:
Update ArcWelderPlugin.py

Issue with subprocess run when cmd_3d_arcs =""

5axes committed:
Update ArcWelder.exe

Update from https://github.com/FormerLurker/ArcWelderLib/releases/tag/1.0.0

5axes committed:
update with

https://github.com/FormerLurker/ArcWelderLib/releases/tag/1.0.0

5axes committed:
Update ArcWelderPlugin.py

5axes committed:
change to active_build_plate_id

On the example of PostProcessingPlugin

5axes committed:
Update ArcWelderPlugin.py
